src/wsproxy.ml

```diff
@@ -45,7 +46,15 @@ let start path handler =
          Lwt_unix.close fd_sock2 >>= fun _ ->
          List.iter (fun fd -> Printf.printf "got fd: %d\n%!" (Obj.magic fd)) newfds;
          Printf.printf "About to fixup the fd\n%!";
-         ignore(handler (Lwt_unix.of_unix_file_descr (List.hd newfds)) msg);
+         Lwt.catch
+          (fun () ->
+            ignore(handler (Lwt_unix.of_unix_file_descr (List.hd newfds)) msg);
+            Lwt.return ()
+          )
```

Comment:
It's annoying here. In fact there should always only be one fd, and handler is closing it... Adding a close for the tail does not harm though.